libs/dal/src/repositories/job/job.repository.ts

```diff
@@ -252,18 +253,22 @@ export class JobRepository extends BaseRepository<JobDBModel, JobEntity, Enforce
     };
   }
 
+  private buildDigestQuery(digestKey: string | undefined, digestValue: string | number | undefined) {
+    const digestQueryV1 = digestKey ? { [`payload.${digestKey}`]: digestValue } : null;
```

Comment:
In v1, we handled digest aggregation by:
1. Extracting a value from the job payload using a specified digest key
2. Checking if another job already existed with the same digest key-value pair
3. If a match was found, we would merge the new job with the existing one, otherwise create new one.